text/0009-golang.md

```diff
@@ -0,0 +1,70 @@
+
+From a sysops perspective, Go containers require less operational tuning compared to the JVM containers as they do not contain a stand-alone virtual machine that requires parametrization.
+
+The Go language is widely used across the industry and open-source communities, especially in the areas related to web and microservices. Popular examples of large open-source projects created with Go include the [Kubernetes](https://github.com/kubernetes/kubernetes) container orchestration platform and the [Docker](https://github.com/docker) container runtime.
+
+The Go project maintains a [list](https://go.dev/solutions/) of case-studies from commercial companies relying on Go in their products and general use-cases for the language.
+
+According to [GitHub statistics](https://madnight.github.io/githut/#/pull_requests/2023/3), Go is the third fastest growing language (behind Python and Java) as measured by the number of new Pull Requests.
+
+## Risks and Drawbacks
+
+Introducing Go as one of the supported backend technologies involves the same general concerns as introducing any new technology into FOLIO. The biggest potential risk is the increased development and maintenance complexity as some new FOLIO backend modules will be created with Go which is not immediately well known to existing FOLIO backend developers that have primarily Java background. As explained in the previous section, however, this risk is contained because of the design of Go which, while being a different language, has many points of similarity to Java.
+
+Another drawback worth mentioning is that a subset of the shared library functionality used by Java modules must be implemented in the form of a shared Go library. The library can, however, be kept simple and include only the most fundamental functionality, similar to [folio-vertx-lib](https://github.com/folio-org/folio-vertx-lib).
+
+Additionally, as history of FOLIO development shows, there exists high level of affinity between specific teams of developers and particular modules and domains. This affinity often results in siloing and can create barriers preventing wider collaboration between various development teams. This siloing exists even within Java-only usage in FOLIO (vert.x vs Spring) because of the high reliance on web programming frameworks. Development in Go has less or no reliance on frameworks -- for instance it's entirely possible to develop FOLIO modules with only Go standard library, Postgres client library and OpenAPI parser library. This more lightweight approach to development can ease some of the cross-module development barriers in FOLIO. 
```

Comment:
I fail to see how introducing a new language, which AFAICT very few Folio developers are familiar with, will help with this "siloing".  Won't this just create a new silo of Go developers who may not be comfortable working with java code, and vice-versa?

Reply:
Yeah at first glance I share @craigmcnally 's feeling on this.  I'd like to understand better from the proposers why they feel the barriers of (say) a Spring developer trying to understand/maintain vert.x code are greater than for Spring developer trying to understand/modify golang code.

Reply:
I agree with the text in the proposal. I find Go more intuitive than any Java framework. It is enticing to develop only based on the standard libraries. This will attract developers and it will also make code reviewing and maintainance easier. I 100% agree with @craigmcnally saying above that the community must find equivalent tools to Sonarqube tests and unit tests which apply to Go. I think this will be a task of the T.C. and will be doable.

Reply:
@craigmcnally I think Ingolf may have said it clearer. The point is that Go relies on frameworks to a much lesser extent than Java and projects in Go usually use the standard lib + popular individual libraries so it tends to be easier to contribute to OSS Go projects. The "siloing" I was referring to is especially pronounced in Java (Spring, JavaEE, vert.x, Quarkus, micronaut, etc)